modules/accredited_representative_portal/app/policies/accredited_representative_portal/power_of_attorney_requests_policy.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+  class PowerOfAttorneyRequestsPolicy
+    def initialize(user, poa_request)
+      @user = user
+      @poa_request = poa_request
+    end
+
+    def pilot_users
+      Settings.accredited_representative_portal.pilot_users.to_h.stringify_keys
+    end
+
+    def authorize
+      return false unless @user
+
+      allowed_codes = Array(pilot_users[@user.email])
+      return false unless allowed_codes
+
+      if @poa_request.is_a?(Array)
+        @poa_request.all? { |request| allowed_codes.include?(request.poa_code) }
+      else
+        allowed_codes.include?(@poa_request.poa_code)
+      end
```

Comment:
What do you think about the following expression that coalesces everything to the one comparison: "user's poa codes must cover records' poa codes"?
```ruby
pilot_user_email_poa_codes =
  Settings
    .accredited_representative_portal
    .pilot_user_email_poa_codes.to_h
    .stringify_keys!

pilot_user_poa_codes = Set.new(pilot_user_email_poa_codes[@user&.email])
poa_requests_poa_codes = Set.new(Array(@record), &:power_of_attorney_code)

pilot_user_poa_codes >= poa_requests_poa_codes
```

Reply:
I like the `all? { include? }` version as well, somewhat a matter of taste since there won't be large enough data sets during the pilot for performance to be impacted by the difference.

Another stylistic thing, I usually like using [`in?` from ActiveSupport](https://api.rubyonrails.org/classes/Object.html#method-i-in-3F) (and reversing operand order accordingly) over `include?`, but again that's just a matter of taste.

Reply:
I like these. I'll make the changes.